src/System.Globalization/tests/CultureInfo/CultureInfoAsync.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class CultureInfoAsync
+    {        
+        [Fact]
+        public void TestCurrentCulturesAsync()
+        {
+            CultureInfo currentCulture = CultureInfo.CurrentCulture;
+            CultureInfo currentUICulture = CultureInfo.CurrentUICulture;
+
+            CultureInfo newCurrentCulture = new CultureInfo(currentCulture.Name.Equals("ja-JP", StringComparison.OrdinalIgnoreCase) ? "en-US" : "ja-JP");
+            CultureInfo newCurrentUICulture = new CultureInfo(currentUICulture.Name.Equals("ja-JP", StringComparison.OrdinalIgnoreCase) ? "en-US" : "ja-JP");
+            
+            CultureInfo.CurrentCulture = newCurrentCulture;
+            CultureInfo.CurrentUICulture = newCurrentUICulture;
+            
+            try
+            {
+                Task t = Task.Run(() => {
+                    Assert.Equal(CultureInfo.CurrentCulture, newCurrentCulture);
+                    Assert.Equal(CultureInfo.CurrentUICulture, newCurrentUICulture);
+                });
+
+                ((IAsyncResult)t).AsyncWaitHandle.WaitOne();
+                t.Wait();                
+            }
+            finally
+            {
+                CultureInfo.CurrentCulture = currentCulture;
+                CultureInfo.CurrentUICulture = currentUICulture;
+            }
+        }
+        
+        [Fact]
+        public void TestCurrentCulturesWithAwait()
+        {
+            CultureInfo currentCulture = CultureInfo.CurrentCulture;
+            CultureInfo currentUICulture = CultureInfo.CurrentUICulture;
+
+            CultureInfo newCurrentCulture = new CultureInfo(currentCulture.Name.Equals("ja-JP", StringComparison.OrdinalIgnoreCase) ? "en-US" : "ja-JP");
+            CultureInfo newCurrentUICulture = new CultureInfo(currentUICulture.Name.Equals("ja-JP", StringComparison.OrdinalIgnoreCase) ? "en-US" : "ja-JP");
+            
+            CultureInfo.CurrentCulture = newCurrentCulture;
+            CultureInfo.CurrentUICulture = newCurrentUICulture;
+            
+            Func<Task> mainAsync = async delegate 
+            {
+                await Task.Yield();
```

Comment:
Ah, looking at this again, I realized that there's one issue.  When running in xunit, xunit has a SynchronizationContext that it uses to ensure at most N tests run at a time.  ```await Task.Yield()``` will post back to the current SynchronizationContext.   That's fine, except the caller is blocking a thread associated with that SynchronizationContext.  Which means that if N == 1 (due to being on a single core box, for example), this will deadlock.  You can fix this by changing this line to be:
```C#
await Task.Delay(1).ConfigureAwait(false);
```
There's a really small chance that the timeout will occur so quickly that this won't actually test what we want it to, but it's unlikely enough that I wouldn't worry about it... on the rare cases where that does happen, the test will still pass.  It's possible to change this in a way that'll guarantee it's testing what we want, but it requires more code.